HW2Question2Updatded.cpp

```diff
@@ -54,6 +55,36 @@ treeNode* lowestCommonAncestor(treeNode *head, int key1, int key2)
     
 }
 
+ treeNode* lowestCommonAncestor(treeNode *head, int key1, int key2)
+{
+    if( key1 == key2)
+    {
+       cout<<"Both keys are the same\n";
+        return nullptr;
+    }
+    
+    else if(checkNodeInTree(head, key1) && checkNodeInTree(head, key1))
+    {
+        return lowestCommonAncestorPrivate(head, key1, key2);
```

Comment:
Minor comment about performance: You are potentially exploring the tree three times, which is two times too many :)
Then again performance is not a problem until it is :)